app/services/metrics/review_turnaround/per_user_project.rb

```diff
@@ -10,22 +11,42 @@ def call
       private
 
       def process
-        filtered_reviews_ids.each_slice(BATCH_SIZE) do |batch|
-          Events::Review.eager_load(:review_request).find(batch).lazy.each do |review|
+        filtered_reviews_ids.lazy.each_slice(BATCH_SIZE) do |batch|
+          Events::Review.eager_load(:review_request).find(batch).each do |review|
             entity = find_user_project(review.owner, review.pull_request.project)
             turnaround = calculate_turnaround(review)
 
-            create_metric(entity, turnaround)
+            create_or_update_metric(entity, turnaround)
           end
         end
+        calculate_avg
       end
 
       def filtered_reviews_ids
         Events::Review.joins(:review_request, owner: :users_projects, pull_request: :project)
```

Comment:
I'm wondering if we are using these join statements, given that we just select the ids.

Reply:
Those joins are just to filter the reviews, like a condition for a Review to be included in the result. Do you think deleting this won't change the result?